lib/spack/docs/docker_for_developers.rst

```diff
@@ -0,0 +1,39 @@
+.. _docker_for_developers:
+
+=====================
+Docker for Developers
+=====================
+
+This guide is intended for people who want to use our prepared docker
+environments to work on developing Spack or working on spack packages. It is
+meant to serve as the companion documentation for the :ref:`packaging-guide`.
+
+--------
+Overview
+--------
+
+To get started, all you need is the latest version of ``docker`` and
+``docker-compose``.
+
+.. code-block:: console
+
+    $ cd share/spack/docker
+    $ docker-compose run --rm ubuntu
+
+This command should drop you into an interactive shell where you can run spack
+within an isolated docker container running ubuntu.  The copy of spack being
+used should be tied to the working copy of your cloned git repo, so any changes
+you make should be immediately reflected in the running docker container.  Feel
+free to add or modify any packages or to hack on spack, itself.  Your contained
+copy of spack should immediately reflect all changes.
+
+To work within a container running a different linux distro, change the "ubuntu"
+argument to any one of the services listed under the ``docker-compose.yml``
+file.
+
+.. code-block:: console
+
+    $ docker-compose config --services
+    fedora
+    ubuntu
+    $ docker-compose run --rm fedora
```

Comment:
somehow I get a
```
Creating network "spack_default" with the default driver
ERROR: 404 page not found
```

at this point :(

Reply:
I'm not sure what the 404 means, exactly, but the version of docker you reported is almost certainly too old.  You will need a relatively recent version of the docker client and server in order for the different containers to work.

For comparison, this is what I get on my Macbook using the new docker for mac engine:

```Bash
 $ docker version
```

```
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

Reply:
Sorry for the long delay and thx for ping-ing!

I updated to
```bash
$ docker --version
Docker version 17.09.0-ce, build afdb6d4

$ docker-compose --version
docker-compose version 1.17.0, build ac53b73
```

now... will report back in a minute :)